src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -822,43 +825,44 @@ public void ProcessTransaction(Transaction transaction, int? blockHeight = null,
         /// Adds a transaction that credits the wallet with new coins.
         /// This method is can be called many times for the same transaction (idempotent).
         /// </summary>
-        /// <param name="transactionHash">The transaction hash.</param>
-        /// <param name="time">The time.</param>
-        /// <param name="isCoinStake">A value indicating whether this is a coin stake transaction or not.</param>
-        /// <param name="index">The index.</param>
-        /// <param name="amount">The amount.</param>
-        /// <param name="script">The script.</param>
+        /// <param name="transaction">The transaction from which details are added.</param>
+        /// <param name="utxo">The unspend output to add to the wallet.</param>
         /// <param name="blockHeight">Height of the block.</param>
         /// <param name="block">The block containing the transaction to add.</param>
-        /// <param name="transactionHex">The hexadecimal representation of the transaction.</param>
         /// <param name="isPropagated">Propagation state of the transaction.</param>
-        private void AddTransactionToWallet(string transactionHex, uint256 transactionHash, uint time, bool isCoinStake, int index, Money amount, Script script,
-            int? blockHeight = null, Block block = null, bool isPropagated = true)
+        private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? blockHeight = null, Block block = null, bool isPropagated = true)
         {
-            this.logger.LogTrace("({0}:'{1}',{2}:'{3}',{4}:{5},{6}:{7},{8}:{9},{10}:{11},{12}:{13})", nameof(transactionHex), transactionHex,
-                nameof(transactionHash), transactionHash, nameof(time), time, nameof(isCoinStake), isCoinStake, nameof(index), index, nameof(amount), amount, nameof(blockHeight), blockHeight);
+            Guard.NotNull(transaction, nameof(transaction));
```

Comment:
This is a private method so using Guard is not so necessary, and you removed the trace (unless it was intentional)